logfire/_internal/utils.py

```diff
@@ -184,3 +190,23 @@ def get_version(version: str) -> Version:
     except ImportError:  # pragma: no cover
         from setuptools._vendor.packaging.version import Version
     return Version(version)  # type: ignore
+
+
+def is_instrumentation_suppressed() -> bool:
+    return bool(
+        context.get_value(SUPPRESS_INSTRUMENTATION_CONTEXT_KEY) or context.get_value(OTEL_SUPPRESS_INSTRUMENTATION_KEY)
+    )
+
+
+@contextmanager
+def suppress_instrumentation():
+    """Context manager to suppress instrumentation from the LLM client."""
```

Comment:
Why not use (and fix / expand if necessary) https://github.com/open-telemetry/opentelemetry-python-contrib/blob/f8758c6902ed725864ff677f739829aa6bce2078/opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py#L188-L192?

Reply:
Some parts of OTEL still use the plain `'suppress_instrumentation'` key, so we need to check/set that too, as you said in https://github.com/pydantic/logfire/issues/135#issuecomment-2098318558.

We could maybe upstream a fix to OTEL to consistently use one key, but we need this to also work for older OTEL versions.

Reply:
We should fix as much of OTEL as we can. But yes the unfortunate reality is that both us and OTEL will have to set and check for any keys ever used.

![image](https://github.com/pydantic/logfire/assets/1755071/11382a9f-67b6-435c-b3ef-a58c13c72d8d)

Reply:
@alexmojaki before you merge this could you address this comment? I'd like to understand why we can't just use the OTEL function or if we can't because it is missing some functionality I'd like to know that there is a plan to upstream it and then replace our function with theirs or something like that.

Reply:
Opened https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2543

Reply:
Great so to make sure I understand we agree that OTEL needs to make some changes that will make future versions work correctly because we do not want to bump our minimum version we are going to do the fixes by exposing our own utility functions.

Will our utility functions be able to to just point to the open telemetry ones in the future?

Reply:
> Will our utility functions be able to to just point to the open telemetry ones in the future?

Sure, although I think it'll be some time before we decide we don't need a fallback for older OTEL versions.